### ISSUE TITLE
Propagate metadata changes for hash-distributed tables with replication factor 1

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#791]
For hash-distributed tables with replication factor 1, we need to propagate the metadata changes to the workers. This includes:
- master_create_distributed_table
- master_create_worker_shards
- DDL on distributed table
- Drop distributed table

**Comment by marcocitus** [comment]
Some things to note:
- In 6.0 we added support for truncate through triggers. We need to explicitly add these triggers as part of the DDL commands that we send to the workers.
- We ran into an issue in MX with the schema being created by super user, which made the schema inaccessible to regular users. We should make sure that the CREATE SCHEMA command specifies an AUTHORIZATION: https://www.postgresql.org/docs/current/static/sql-createschema.html
- When creating a repmodel 's' table, we should make sure it does not get co-located with a 'c' table and vice-versa. Otherwise we might get strange behaviour around foreign key references on MX tables pointing to non-MX tables.
